### ISSUE TITLE
Make a REPL for DSpace using JShell

### ISSUE THREAD

**Issue by hardyoyo** [DSpace/DSpace#9361]
**Is your feature request related to a problem? Please describe.**

USE CASE:
As a developer new to DSpace, I often find myself needing to quickly test code snippets, debug issues, or explore the DSpace API without the overhead of setting up a full development environment. The current development process lacks a straightforward way to interactively test code, which can slow down the development cycle and increase the learning curve for new contributors.

WHAT IS A REPL?

REPL is short for [Read-Eval-Print Loop](https://en.wikipedia.org/wiki/Read%E2%80%93eval%E2%80%93print_loop), they are common in many languages and frameworks (Python, Ruby, Rails). [JShell](https://docs.oracle.com/en/java/javase/20/jshell/introduction-jshell.html) is a Read-Eval-Print Loop (REPL) tool first introduced in Java 9. JShell allows you to execute Java code snippets and get immediate feedback. It's useful for trying out new Java features, testing small code snippets, or learning Java. 

**Describe the solution you'd like**

Create a dspace-shell script in the bin folder of the dspace project. This script will start up jshell with the dspace/lib folder in the classpath, and also load a jshell startup command to pre-load all the DSpace libraries needed to provide a REPL environment for DSpace. The Jshell startup script should probably live in the dspace/config folder.

**Describe alternatives or workarounds you've considered**
We can continue without a REPL, but it would be a nice thing to have. 

**Additional context**

WHY A REPL FOR DSPACE?

1. **Immediate Feedback:** Get instant results and feedback while experimenting with DSpace code.
2. **Rapid Prototyping:** Quickly test and prototype code snippets without the need for a full program structure.
3. **Learning and Teaching:** Ideal for learning DSpace coding practices or teaching concepts due to its interactive nature.
4. **Debugging Aid:** Use it to isolate and debug specific code segments on the fly.
5. **API Exploration:** Easily explore Java APIs and test library functionalities interactively.
6. **Experimentation:** Great for trying out new language features or syntax without the overhead of a complete project. Or experiment with an unfamiliar dependency, before you add it to the pom.xml
7. **Scripting:** Execute scripts and automate simple tasks without the need for a full-fledged application.
8. **Curation:** A dreamy environment for interactively drafting a DSpace curation script.

**Comment by hardyoyo** [comment]
TLDR: the proposal is to add two shell scripts. Java already comes with Jshell.

**Comment by MW3000** [comment]
@hardyoyo I love the idea. I was just trying to get JShell working with DSpace. But regardless where I point the classpath, I don’t seem to have access to the DSpace libraries. My best guess was:

`jshell --class-path [dspace-backend-installation]/lib`

but without success. Did you manage to use JShell with DSpace? What did you specify as class-path?

Adding `[dspace-backend-installation]/lib` as `module-path` ended in several error messages.

**Comment by hardyoyo** [comment]
Hi, I never did get this working. I suspect DSpace may no longer be an app you can do this for. But someone might have better luck than I.
